plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/fs/writeFile.ts

```diff
@@ -56,7 +56,15 @@ export function createWriteFileAction() {
         ctx.input.path,
       );
 
-      fs.outputFileSync(destFilepath, ctx.input.content);
+      let formattedContent = ctx.input.content;
+      try {
```

Comment:
You could pass a flag to allow users to specify this behaviour to make it non-breaking.

Reply:
@sblausten i made the processing as optional. With this modification, users can now include a preserveFormatting boolean flag in the input schema. If set to true, the content will be formatted as JSON before writing it to the file, preserving its structure. If set to false or omitted, the content will be written as is without any formatting changes.

Reply:
That's great thank!